astropy/io/fits/convenience.py

```diff
@@ -460,19 +460,30 @@ def table_to_hdu(table):
     """
     # Avoid circular imports
     from .connect import is_column_keyword, REMOVE_KEYWORDS
+    
+    # Header to store Time related metadata
+    hdr = None
 
     # Not all tables with mixin columns are supported
     if table.has_mixin_columns:
         # Import is done here, in order to avoid it at build time as erfa is not
         # yet available then.
         from ...table.column import BaseColumn
+        from ...table import QTable, Column
+        from ...time import Time
+        from .time import replace_time_table
 
         # Only those columns which are instances of BaseColumn or Quantity can be written
-        unsupported_cols = table.columns.not_isinstance((BaseColumn, Quantity))
+        unsupported_cols = table.columns.not_isinstance((BaseColumn, Quantity, Time))
         if unsupported_cols:
             unsupported_names = [col.info.name for col in unsupported_cols]
             raise ValueError('cannot write table with mixin column(s) {0}'
                          .format(unsupported_names))
+        
+        time_cols = table.columns.isinstance(Time)
+        if time_cols:
+            newtable = QTable(table)
```

Comment:
I think it should be possible to make the second table at very little cost of additional memory by using `copy=False`.  Even the `Time.jd1` and `jd2` can be put in as columns that are just views of the original data.

Interestingly I have been independently thinking about the same idea (intermediate table) for serializing `Time` and `SkyCoord` to ECSV!